genie-web/src/main/java/com/netflix/genie/web/services/impl/LocalJobKillServiceImpl.java

```diff
@@ -165,6 +187,17 @@ public void killJob(
         }
     }
 
+    private void handleV4JobKill(final String id) throws GenieException {
+        if (this.jobKillCallbacks.containsKey(id)) {
+            this.jobKillCallbacks.get(id).run();
+        } else {
+            log.error("Job not killed. No callback found for killing the job with id - {} ", id);
+            throw new GenieException(
+                HttpStatus.BAD_REQUEST.value(),
```

Comment:
1. There's already a `GenieBadRequestException` you can use rather than hand rolling your own code here
2. I don't think this would be a `BadRequest` as that is a 4xx code which is a user error. In this case it's likely not a user error. If they request to kill a job that is already killed or can't be found we may just want to swallow it. Otherwise i would think it's some sort of `InternalServerError`